### PR TITLE
Design Refactor to use constructors, pkg-pvt get methods in response related objects

### PR DESCRIPTION
**Pull request by scottyplunkett**
- Removed extraneous directories from repo
- Added constructors, public get methods, and private build methods to response related objects
- Cleaned and clarified test

### CODE REVIEW

**Comment by scottyplunkett** [comment]
Hopefully this should help to shrink future pull request. Sorry this is such a big one!

File: src/com/scottyplunkett/server/DirectoryContent.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    String get(){
+        return build;
+    }
+
+    private String build(String directory) {
+        String content = "";
+        File[] files = new File(directory).listFiles();
+        List<String> names =  Arrays.asList(files).parallelStream()
```
Comment (esmevane) [inline src/com/scottyplunkett/server/DirectoryContent.java]: 👏

File: src/com/scottyplunkett/server/DirectoryContent.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+                                    .map(file -> file.getName())
+                                    .collect(Collectors.toList());
+        for (String name : names) {
```
Comment (esmevane) [inline src/com/scottyplunkett/server/DirectoryContent.java]: I think there's a reducer / fold operation you can use here to get these rolled up into a single string (using a `StringBuilder`?), might be worth looking into.

File: src/com/scottyplunkett/server/HTTPRequest.java [hunk 3]
```diff
@@ -5,20 +5,27 @@
 import java.util.ArrayList;
 
 class HTTPRequest {
-    static ArrayList<String> read(BufferedReader in) throws IOException {
+    private ArrayList<String> read;
+    private String requestLine;
+
+    HTTPRequest(BufferedReader in) throws IOException {
+        read = new ArrayList<>();
+        read(in);
```
Comment (esmevane) [inline src/com/scottyplunkett/server/HTTPRequest.java]: The name collision here (`read` the method, `read` the private list) was a little confusing for me.

File: src/com/scottyplunkett/server/PageContent.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+import java.io.UnsupportedEncodingException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.stream.Collectors;
+
+class PageContent {
+    private String build;
+
+    PageContent(Path pagesPath, String requested) throws IOException {
+        build = build(pagesPath, requested);
```
Comment (esmevane) [inline src/com/scottyplunkett/server/PageContent.java]: I'm seeing the same variable / method name pattern here.  Maybe you're looking for an internal variable name that has a clear connection to the operation creating it?  Some ideas:

* `read` / `read` = `requestContent` / `readRequestContent`
* `build` / `build` = `pageContent` / `buildPageContent`

Basically, `theValueName` / `theProcessName`.

File: src/com/scottyplunkett/server/Parser.java [hunk 5]
```diff
@@ -2,37 +2,19 @@
...
 import java.util.function.Function;
 
 class Parser {
-    static Function<String, String[]> splitBySpace = string -> string.split("\\s+");
-    static Function<String, String[]> splitByQuestionMark = string -> string.split("\\?");
-
-    static String parseFor(String parseType, String getRequest) {
-        return findRequestMethod(getRequest);
-    }
+    private static final Function<String, String[]> splitBySpace = string -> string.split("\\s+");
```
Comment (esmevane) [inline src/com/scottyplunkett/server/Parser.java:8]: Nice!

**Review by esmevane — APPROVED** [review]
This is good stuff!  I like the functional notation on later objects, that's 👍 

Great work!

File: src/com/scottyplunkett/server/HTTPResponse.java [hunk 6]
```diff
@@ -3,21 +3,26 @@
 import java.io.IOException;
 
 class HTTPResponse {
-    static String body;
+    private String build;
 
-    static void setBody(String htmlString) {
-        body = htmlString;
+    HTTPResponse(String requested) throws IOException {
```
Comment (NicoleCarpenter) [inline src/com/scottyplunkett/server/HTTPResponse.java]: 🆒

**Review by NicoleCarpenter — APPROVED** [review]
I don't have anything to add beyond what Josh suggested. This was a good refactor 👍 Approved once that feedback is addressed.

File: src/com/scottyplunkett/server/DirectoryContent.java (hunk 2 above)
Comment (scottyplunkett) [reply src/com/scottyplunkett/server/DirectoryContent.java]: I will definitely look into that. Thanks!

File: src/com/scottyplunkett/server/HTTPRequest.java (hunk 3 above)
Comment (scottyplunkett) [reply src/com/scottyplunkett/server/HTTPRequest.java]: I could see how that could be confusing. My goal was to create classes with phrasing that represented both the noun (the Value), and the verb (the Process) for ease of understanding.
ex.
Issue the command: Scott, read (verb) the book. 
Now get the value: Scott, give me your read. (noun).
ex. 
Scott, build (action) the building. 
Scott, show us the build (value).

I can see how this is less clarifying than I intended, and if it's confusing at first glance than I should change this templating syntax to be more expressive of the implementation as you've suggested below.

Comment (esmevane) [reply src/com/scottyplunkett/server/HTTPRequest.java]: I follow you - I think the problem here is that I had two concepts in my head.  The overloading or shadowing of a term is something a lot of languages handle differently.  So when I looked at it, for a minute, I was wondering, "Is this a special operation that a `ArrayList` has in Java, where you can call it like a function?"

**Review by esmevane — APPROVED** [review]

**Pull request merged**